### PR TITLE
core/vm: more benchmarks for bls g1/g2-multiexp precompiles

### PR DESCRIPTION
**Pull request by jwasinger**
Currently, our benchmark set only measure several input sizes for msm and pairing.  This introduces more comprehensive benchmarks with a wider-variety of input sizes.

For pairing, inputs are random points (which is the worst-case for gnark's implementation).  For msm, inputs are random points and scalars whose binary representation is all 1s (`0xfff..fff`)

### CODE REVIEW

**Comment by jwasinger** [comment]
I need to fill in gas values as well...

**Comment by jwasinger** [comment]
If we want, I can include the generator for these (and the others I am about to add) as part of this PR.

**Comment by jwasinger** [comment]
So actually, the only precompiles we are really lacking benchmark coverage for are MSM.  Previous to this we only benchmark the following input pair counts: 1, 3, 16 25.

The new benchmarks cover pair counts 1..32, 64, 128 , 512, 1024, 2048 and 4877 (the size of the fuzzer-found worst-case input that fits in a 10mil gas block).

**Comment by holiman** [comment]
> The new benchmarks cover pair counts 1..32, 64, 128 , 512, 1024, 2048 and 4877 (the size of the fuzzer-found worst-case input that fits in a 10mil gas block).

Not sure I understand. For G1, As far as I can tell, the new tests include first of all these three

```
BenchmarkPrecompiledBLS12381G2MultiExp/bls_g2multiexp_single-Gas=54000-8                     862           1378579 ns/op             54000 gas/op               39.16 mgas/s       44727 B/op        142 allocs/op
BenchmarkPrecompiledBLS12381G2MultiExp/bls_g2multiexp_multiple-Gas=103140-8                  622           1885135 ns/op            103140 gas/op               54.70 mgas/s       46242 B/op        152 allocs/op
BenchmarkPrecompiledBLS12381G2MultiExp/bls_g2multiexp_larger-Gas=335250-8                    133           9249940 ns/op            335250 gas/op               36.23 mgas/s       62751 B/op        245 allocs/op
```
THen, it's `0-100` of these: 
```
BenchmarkPrecompiledBLS12381G2MultiExp/matter_g2_multiexp_0-Gas=240480-8                     171           7028747 ns/op            240480 gas/op               34.20 mgas/s       56312 B/op        233 allocs/op
BenchmarkPrecompiledBLS12381G2MultiExp/matter_g2_multiexp_1-Gas=240480-8                     170           7087736 ns/op            240480 gas/op               33.92 mgas/s       56250 B/op        225 allocs/op
BenchmarkPrecompiledBLS12381G2MultiExp/matter_g2_multiexp_2-Gas=240480-8                     171           6827471 ns/op            240480 gas/op               35.21 mgas/s       56295 B/op        231 allocs/op

```
That is: 100 tests with the same exact input sizes, and gas usage. Is that intentional? 
It's a bit hard to follow. Github won't show me the json diff (too large), and checking it locally in e.g. `gitk` is hard, because it's spread over 10 commits. Perhaps squash it?

**Comment by holiman** [comment]
> For G1, As far as I can tell, 

Scratch that, my editor was confusing me. LGTM

**Review by holiman — APPROVED** [review]
LGTM
